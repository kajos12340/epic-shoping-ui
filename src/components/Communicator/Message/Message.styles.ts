import styled from 'styled-components';

export const Container = styled.div<{ isOwn: boolean }>`
  display: flex;
  justify-content: ${({ isOwn }) => (isOwn ? 'flex-end' : 'flex-start')};
  
  & > div {
    max-width: 80%;
  }
 
  margin-bottom: 15px;
`;

export const Content = styled.div`
  border: solid 2px #00796b;
  border-radius: 10px;
  display: grid;
  padding: 10px;
  
  grid-template: "icon date" auto
                 "icon text" auto
                 / auto 1fr;
  grid-gap: 10px 10px;
`;

export const OwnContent = styled(Content)`
  grid-template: "date icon" auto
                 "text icon" auto
                 / 1fr auto;
`;

export const Icon = styled.div`
  grid-area: icon;
  display: flex;
  justify-content: center;
  align-items: center;
  
  height: 50px;
  width: 50px;
  background-color: #00796b;
  border-radius: 100%;
  
  color: white;
  font-size: 1.2em;
  
  text-transform: capitalize;
`;

export const Date = styled.div`
  grid-area: date;
  font-size: 0.85em;
  font-weight: 500;
  color: #676767;
`;

export const Text = styled.div`
  grid-area: text;
`;
