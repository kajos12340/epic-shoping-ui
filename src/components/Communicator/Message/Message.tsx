import React from 'react';

import {
  Content, Container, Icon, OwnContent, Date, Text,
} from './Message.styles';

export interface IMessage {
  author: {
    login: string,
    _id: string,
    color: string,
  },
  isOwn?: boolean,
  text: string,
  date: string,
}

interface IMessageProps extends IMessage{
  isOwn: boolean,
}

const Message = ({
  author, isOwn, text, date,
}: IMessageProps) => {
  const ContentWrapper = isOwn ? OwnContent : Content;

  return (
    <Container isOwn={isOwn}>
      <ContentWrapper bgColor={`#${author.color}`}>
        <Icon bgColor={`#${author.color}`}>
          {author.login[0]}
        </Icon>
        <Date>{`${author.login} - ${date}`}</Date>
        <Text>{text}</Text>
      </ContentWrapper>
    </Container>
  );
};

export default Message;
