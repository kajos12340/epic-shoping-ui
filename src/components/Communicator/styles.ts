import styled from 'styled-components';
import { Paper as PaperDefault } from '@material-ui/core';

// eslint-disable-next-line import/prefer-default-export
export const Container = styled(PaperDefault)`
  height: 72vh !important;
`;

export const Content = styled.div`
  height: 100%;
  display: grid;
  grid-template-rows: 1fr auto;
`;

export const Messages = styled.div`
  margin-bottom: 12px;
  grid-template-rows: 1fr auto;
  overflow-y: auto;
`;

export const Input = styled.div`
  display: flex;
  grid-template-rows: 1fr auto;
  
  & > button {
    min-width: 15%;
  }
`;
