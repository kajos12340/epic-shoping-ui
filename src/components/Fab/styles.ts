import styled from 'styled-components';
import FabDefault from '@material-ui/core/Fab';

// eslint-disable-next-line import/prefer-default-export
export const PositionedFab = styled(FabDefault)`
  position: fixed !important;
  bottom: 5%;
  right: 5%;
  
  @media (max-width: 960px) {
    bottom: 10%;
    right: 3%;
  }
`;
