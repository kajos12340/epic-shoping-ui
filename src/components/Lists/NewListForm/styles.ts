import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const ButtonContainer = styled.div`
  margin: 5px 0;
  display: flex;
  justify-content: flex-end;
`;
