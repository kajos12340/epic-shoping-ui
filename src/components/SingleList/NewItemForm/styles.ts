import styled from 'styled-components';

export const Form = styled.form`
  display: grid;
  grid-template: "name quantity unit buttons" auto / 50% 20% 20% 10%;
  grid-column-gap: 5px;
  margin: 15px 0;
  
  & > div {
    height: 100%;
    width: 100%;
    align-self: start;
    
    & > div {
      margin: 0 !important; 
    }
  }
  
  @media (max-width: 960px) {
    grid-template: "name quantity unit buttons" auto / 50% 20% 20% 10%;
    grid-template: "name name buttons" auto
                   "quantity unit buttons" auto
                    / 40% 40% 20%;                
    grid-gap: 15px 5px;
  }
`;

export const Name = styled.div`
  grid-area: name;
`;

export const Quantity = styled.div`
  grid-area: quantity;
`;

export const Unit = styled.div`
  grid-area: unit;
`;

export const Buttons = styled.div`
  grid-area: buttons;
  display: flex;
  align-items: flex-start;
  justify-content: center;
  
  & > button {
    padding: 3px 0 0 0;
  }
  
  & svg {
    font-size: 2em;
  }
  
  @media (max-width: 960px) {
    align-items: center; 
  }
`;
