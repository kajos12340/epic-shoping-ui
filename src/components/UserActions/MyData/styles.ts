import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Container = styled.section`
  margin-bottom: 30px;
  display: grid;
  grid-template: "logo login"
                 "logo email"
                 "logo lastLoginDate"
                 "logo registerDate"
                 / 20% calc(80% - 20px);
  grid-column-gap: 20px;
  grid-row-gap: 10px;
  
  & > div {
    word-break: break-word;
    display: flex;
    flex-direction: row;
  }
  
  @media (max-width: 600px) {
    grid-template: "logo"
                 "login"
                 "email"
                 "lastLoginDate"
                 "registerDate"
                 / 100%;
                 
    & > div {
      word-break: break-word;
      display: flex;
      flex-direction: column;
    }                 
  }
`;

export const Logo = styled.div`
  grid-area: logo;
  & svg {
    font-size: 6em;
  }
`;

const InfoRow = styled.div`
  &>div:first-child {
    color: #00796b;
    font-weight: 500;
    font-size: 95%;
    margin-right: 7px;
  }
`;

export const Login = styled(InfoRow)`
  grid-area: login;
`;

export const Email = styled(InfoRow)`
  grid-area: email;
`;

export const LastLoginDate = styled(InfoRow)`
  grid-area: lastLoginDate;
`;

export const RegisterDate = styled(InfoRow)`
  grid-area: registerDate;
`;
