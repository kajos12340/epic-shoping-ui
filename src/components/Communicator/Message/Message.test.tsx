import React from 'react';
import { shallow } from 'enzyme';

import { Icon, Container } from './Message.styles';
import Message, { IMessageProps } from './index';

describe('Message', () => {
  const createWrapper = ({
    isOwn, author, text, date,
  }: IMessageProps) => shallow(
    <Message isOwn={isOwn} author={author} text={text} date={date} />,
  );

  it('should match snapshot', () => {
    const wrapper = createWrapper({
      isOwn: true,
      text: 'test message',
      date: '12:33 24.05.2020',
      author: {
        _id: 'testId',
        color: '123456',
        login: 'testLogin',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should generate icon in proper color', () => {
    const bgColor = '123456';
    const wrapper = createWrapper({
      isOwn: true,
      text: 'test message',
      date: '12:33 24.05.2020',
      author: {
        _id: 'testId',
        color: bgColor,
        login: 'testLogin',
      },
    });

    const icon = wrapper.find(Icon);
    expect(icon.prop('bgColor')).toBe(`#${bgColor}`);
  });

  it('should proper style own messages', () => {
    const bgColor = '123456';
    const wrapper = createWrapper({
      isOwn: true,
      text: 'test message',
      date: '12:33 24.05.2020',
      author: {
        _id: 'testId',
        color: bgColor,
        login: 'testLogin',
      },
    });

    const container = wrapper.find(Container);
    expect(container.prop('isOwn')).toBeTruthy();
  });
});
