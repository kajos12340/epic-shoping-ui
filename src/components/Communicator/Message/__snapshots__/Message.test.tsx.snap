// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
<styled.div
  isOwn={true}
>
  <Styled(styled.div)
    bgColor="#123456"
  >
    <styled.div
      bgColor="#123456"
    >
      t
    </styled.div>
    <styled.div>
      testLogin - 12:33 24.05.2020
    </styled.div>
    <styled.div>
      test message
    </styled.div>
  </Styled(styled.div)>
</styled.div>
`;
